server/scripts/import-tests/index.js

```diff
@@ -159,277 +151,206 @@ const ariaat = {
+                  parsedTest
+              });
+        return testPlan.id;
     },
 
     /**
-     * Gets the test id and adds the test record if it does not exist
-     * @param {string} testFullName - the name of the test
-     * @param {string} file - the relative path to the test in the repo
-     * @param {int} exampleID - foreign key into the apg_example table
-     * @param {int} testVersionID - foreign key into the test_version table
-     * @param {int} executionOrder - order of the tests (within APG pattern)
-     * @return {int} id
+     * Checks TestPlan.parsedTest.testActions to see if it has the relevant test actions to run the test and inserts it if not
+     * @param {string} testName - the name of the test
+     * @param {string} file - the relative path to the test file in the repository (ideally {@link https://github.com/w3c/aria-at.git})
+     * @param {number} testPlanId - TestPlan.id to be queried to update the TestPlan.parsedTest.testActions if necessary
+     * @param {string} commitHash - the hash of the latest version of tests pulled from the repository (ideally {@link https://github.com/w3c/aria-at.git})
+     * @param {string} commitMsg - the message of the latest version of tests pulled from the repository (ideally {@link https://github.com/w3c/aria-at.git})
+     * @param {number} executionOrder - the order in which the test step is executed (within the APG pattern)
+     * @returns {number | null} - returns TestPlan.id
      */
-    async upsertTest(
-        testFullName,
+    async upsertTestPlanParsedTestActions(
+        testName,
         file,
-        exampleID,
-        testVersionID,
+        testPlanId,
+        commitHash,
+        commitMsg,
         executionOrder
     ) {
-        const testResult = await client.query(
-            'SELECT id FROM test WHERE file=$1 AND test_version_id=$2',
-            [file, testVersionID]
+        const testPlanResult = await client.query(
+            'SELECT id, "parsedTest" FROM "TestPlan" WHERE id=$1 AND revision=$2',
+            [testPlanId, commitHash]
         );
-        let testID = testResult.rowCount ? testResult.rows[0].id : undefined;
-        if (!testID) {
-            testID = await this.insertRowReturnId(
-                'INSERT INTO test(name, file, apg_example_id, test_version_id, execution_order) VALUES($1, $2, $3, $4, $5) RETURNING id',
-                [testFullName, file, exampleID, testVersionID, executionOrder]
+        let testPlan = testPlanResult.rowCount ? testPlanResult.rows[0] : null;
+
+        // check to see if test action object already exists in testActions dataset
+        if (testPlan) {
+            const testStepsFound = testPlan.parsedTest.testActions.find(
+                test => test.executionOrder === executionOrder
             );
+            // short circuit method because parsedTest.testActions.[action] is already present
+            if (testStepsFound) return testPlan.id;
```

Comment:
Instead of `testActions` can I suggest `commands`? That matches a convention Simon and I sort of settled on a couple months ago.

I think in this case `command` is better than `testCommand` I want to avoid the trap of putting test in front of every variable name, haha, except in the case where we're too late as in TestPlan.

Reply:
Doesn't `commands` in their context refer to the actual keyboard input commands being done?

Reply:
Yes, in the current code, commands sometimes refers to the keys, but that's definitely something I want to change. And it's not just keys, it will encompass gestures on mobile devices too. In the CG when we touched on this terminology problem, the term "input" came up, which I think fits really well. So I want to gradually rename anything related to input to the name input, and I want to save the term command for the case we're dealing with here, the statements that make up a single test.

input -> the keyboard key or gesture
command -> columns in the test authoring CSV that contains either an instruction or an assertion

Reply:
If I'm understanding correctly, I think these are just `tests`: https://github.com/w3c/aria-at/wiki/Working-Mode#definitions

The hierarchy is: Test Plan ("checkbox") > Test ("Navigate to an unchecked checkbox in reading mode	") > Commands ("X / Shift+X") > Assertion ("Role 'checkbox' is conveyed")

Reply:
That's right Seth - Howard and I talked about it yesterday and that was the conclusion.

Reply:
sounds good! And seems like it might have implications on the name of the field as well? `TestPlan.parsedTest` should probably be closer to `TestPlan.parsed` since the JSON object is referring to the TestPlan as a whole, right?

Reply:
Yes, that's right.